src/components/TaskItem.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Task } from './types';
+
+interface IProps {
+  selectedTask: Task | undefined;
+  startEditing: (id: string) => void;
+  editTask: () => void;
+  deleteTask: (id: string) => void;
+  checkTask: (id: string) => void;
+  task: Task;
+}
+
+function TaskItem(props: IProps) {
+  const { selectedTask, startEditing, editTask, deleteTask, checkTask, task } =
```

Comment:
O linter quebrou isso assim ? Acho que  algo assim ficaria melhor
```suggestion
  const { 
      selectedTask, 
      startEditing, 
      editTask, 
      deleteTask, 
      checkTask, 
      task 
} = props;
```

Reply:
Foi ele que quebrou assim, achei estranho também e toda formatação que fazia ele estava barrando, aí acabei deixando.